tasks/eth/010-1-guardian-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,217 @@
+
+            // Assert that only the expected accounts have been written to.
+            for (uint256 j; j < accountAccess.storageAccesses.length; j++) {
+                Vm.StorageAccess memory storageAccess = accountAccess.storageAccesses[j];
+                if (storageAccess.isWrite) {
+                    address account = storageAccess.account;
+                    require(
+                        // We set the guardian slot on the Superchain Config.
+                        account == address(proxies.SuperchainConfig)
+                        // State changes the Safe's are also expected.
+                        || account == address(proxyAdminOwnerSafe) || account == address(securityCouncilSafe)
+                            || account == address(foundationUpgradesSafe),
+                        "state-100"
+                    );
+                }
+            }
+        }
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck(Vm.AccountAccess[] memory accesses, SimulationPayload memory /* simPayload */ )
+        internal
+        override
+    {
+        console.log("Running post-deploy assertions");
+
+        checkStateDiff(accesses);
+        checkSemvers();
+        checkSuperchainConfig();
+        checkProxyAdminOwnerSafe();
+
+        console.log("All assertions passed!");
+    }
+
+    function getCodeExceptions() internal view override returns (address[] memory) {
+        // Safe owners will appear in storage in the LivenessGuard when added, and they are allowed
+        // to have code AND to have no code.
+        address[] memory securityCouncilSafeOwners = securityCouncilSafe.getOwners();
+
+        // To make sure we probably handle all signers whether or not they have code, first we count
+        // the number of signers that have no code.
+        uint256 numberOfSafeSignersWithNoCode;
+        for (uint256 i = 0; i < securityCouncilSafeOwners.length; i++) {
+            if (securityCouncilSafeOwners[i].code.length == 0) {
+                numberOfSafeSignersWithNoCode++;
+            }
+        }
+
+        // Then we extract those EOA addresses into a dedicated array.
+        uint256 trackedSignersWithNoCode;
+        address[] memory safeSignersWithNoCode = new address[](numberOfSafeSignersWithNoCode);
+        for (uint256 i = 0; i < securityCouncilSafeOwners.length; i++) {
+            if (securityCouncilSafeOwners[i].code.length == 0) {
+                safeSignersWithNoCode[trackedSignersWithNoCode] = securityCouncilSafeOwners[i];
+                trackedSignersWithNoCode++;
+            }
+        }
+
+        // And finally, we set the Safe signer exceptions.
+        address[] memory shouldHaveCodeExceptions = new address[](numberOfSafeSignersWithNoCode);
+        for (uint256 i = 0; i < safeSignersWithNoCode.length; i++) {
+            shouldHaveCodeExceptions[i] = safeSignersWithNoCode[i];
+        }
+
+        return shouldHaveCodeExceptions;
+    }
```

Comment:
I think we can remove this whole function?

Reply:
Ah you're right, done in 1bf8a4d